toolkit/smart-contracts/offchain/src/governed_map/mod.rs

```diff
@@ -0,0 +1,164 @@
+use crate::csl::{
+	empty_asset_name, get_builder_config, unit_plutus_data, CostStore, Costs,
+	TransactionBuilderExt, TransactionExt,
+};
+use crate::governance::GovernanceData;
+use crate::multisig::submit_or_create_tx_to_sign;
+use crate::multisig::MultiSigSmartContractResult::TransactionSubmitted;
+use crate::plutus_script::PlutusScript;
+use crate::{
+	await_tx::{AwaitTx, FixedDelayRetries},
+	cardano_keys::CardanoPaymentSigningKey,
+	csl::TransactionContext,
+	multisig::MultiSigSmartContractResult,
+};
+use anyhow::anyhow;
+use cardano_serialization_lib::{PlutusData, Transaction, TransactionBuilder};
+use ogmios_client::{
+	query_ledger_state::{QueryLedgerState, QueryUtxoByUtxoId},
+	query_network::QueryNetwork,
+	transactions::Transactions,
+	types::OgmiosUtxo,
+};
+use partner_chains_plutus_data::governed_map::{
+	governed_map_datum_to_plutus_data, GovernedMapDatum,
+};
+use sidechain_domain::byte_string::ByteString;
+use sidechain_domain::{PolicyId, UtxoId};
+
+pub async fn run_insert<
+	C: QueryLedgerState + QueryNetwork + Transactions + QueryUtxoByUtxoId,
+	A: AwaitTx,
+>(
+	genesis_utxo: UtxoId,
+	key: String,
+	value: ByteString,
+	payment_signing_key: &CardanoPaymentSigningKey,
+	ogmios_client: &C,
+	await_tx: &A,
+) -> anyhow::Result<Option<MultiSigSmartContractResult>> {
+	let ctx = TransactionContext::for_payment_key(payment_signing_key, ogmios_client).await?;
+	let (validator, policy) = crate::scripts_data::governed_map_scripts(genesis_utxo, ctx.network)?;
+	let validator_address = validator.address_bech32(ctx.network)?;
+	let validator_utxos = ogmios_client.query_utxos(&[validator_address]).await?;
+
+	let tx_hash_opt = match get_current_value(validator_utxos, key.clone(), policy.policy_id())? {
+		Some(current_value) if current_value != value => {
+			return Err(anyhow!(
+				"There is already a value stored for key '{}': {:?}",
+				key,
+				current_value
```

Comment:
```suggestion
				"There is already a value stored for key '{key}': {current_value:?}"
```
this little trick to get cargo fmt to act normal for once